client/ayon_core/pipeline/ayon_uri.py

```diff
@@ -0,0 +1,298 @@
+            product=product,
+            version=version,
+            representation=representation_name
+        )
+    )
+
+
+def get_representation_by_names(
+        project_name: str,
+        folder_path: str,
+        product_name: str,
+        version_name: str,
+        representation_name: str,
+) -> Optional[dict]:
+    """Get representation entity for asset and subset.
+
+    If version_name is "hero" then return the hero version
+    If version_name is "latest" then return the latest version
+    Otherwise use version_name as the exact integer version name.
+
+    """
+
+    if isinstance(folder_path, dict) and "name" in folder_path:
+        # Allow explicitly passing asset document
+        folder_entity = folder_path
+    else:
+        folder_entity = get_folder_by_path(project_name,
+                                           folder_path,
+                                           fields=["id"])
+    if not folder_entity:
+        return
+
+    if isinstance(product_name, dict) and "name" in product_name:
+        # Allow explicitly passing subset document
+        product_entity = product_name
+    else:
+        product_entity = get_product_by_name(project_name,
+                                             product_name,
+                                             folder_id=folder_entity["id"],
+                                             fields=["id"])
+    if not product_entity:
+        return
+
+    if version_name == "hero":
+        version_entity = get_hero_version_by_product_id(
+            project_name,
+            product_id=product_entity["id"])
+    elif version_name == "latest":
+        version_entity = get_last_version_by_product_id(
+            project_name,
+            product_id=product_entity["id"])
+    else:
+        version_entity = get_version_by_name(project_name,
+                                             version_name,
+                                             product_id=product_entity["id"])
+    if not version_entity:
+        return
+
+    return get_representation_by_name(project_name,
+                                      representation_name,
+                                      version_id=version_entity["id"])
+
+
+def get_representation_path_by_names(
+        project_name: str,
+        folder_path: str,
+        product_name: str,
+        version_name: str,
+        representation_name: str) -> Optional[str]:
+    """Get (latest) filepath for representation for folder and product.
+
+    See `get_representation_by_names` for more details.
+
+    Returns:
+        str: The representation path if the representation exists.
+
+    """
+    representation = get_representation_by_names(
+        project_name,
+        folder_path,
+        product_name,
+        version_name,
+        representation_name
+    )
+    if representation:
+        path = get_representation_path(representation)
+        return path.replace("\\", "/")
+
+
+def get_representation_path_by_ayon_uri(
```

Comment:
Function `get_representation_path_by_ayon_uri` should not expect context at all. From API point of view it is very confusion. Uri is all the function needs to return representation entity from database. The context is needed for special case, which should be handled in different function (that can use this one).